libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -933,6 +933,25 @@ string YulUtilFunctions::calldataArrayIndexAccessFunction(ArrayType const& _type
 	});
 }
 
+string YulUtilFunctions::calldataArrayIndexRangeAccess(ArrayType const& _type)
+{
+	solAssert(_type.dataStoredIn(DataLocation::CallData), "");
+	string functionName = "calldata_array_index_range_access_" + _type.identifier();
+	return m_functionCollector.createFunction(functionName, [&]() {
+		return Whiskers(R"(
+			function <functionName>(offset, length, startIndex, endIndex) -> offsetOut, lengthOut {
+				if iszero(lt(endIndex, length)) { invalid() }
+				if iszero(lt(0, startIndex)) { invalid() }
```

Comment:
``startIndex`` *can* be ``0``, can't it? You can have a look at ``ExpressionCompiler::visit(IndexRangeAccess)`` for a reference implementation of the tests. As a matter of fact you can replace the respective ``appendInlineAssembly`` there with a ``callYulFunction`` of this implementation - that deduplicates things and will immediately increase the test coverage.

Reply:
`startIndex` can be `0`, it was my bad, so I changed that. also did `callYulFunction` instead of `appendInlineAssembly` in `ExpressionCompiler::visit(IndexRangeAccess)`